CH-675/CH-676 added Entities to Order Service

NatalliaBulyha opened the pull request:
jira [create tables and entities + add data] https://cozyhometeam.atlassian.net/browse/CH-676?atlOrigin=eyJpIjoiZmI3NTE1NWQ0OThlNDAxOWIyZmE1NDA4MjFjYWY2OWYiLCJwIjoiaiJ9

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/model/OrderDetails.java:
```diff
@@ -0,0 +1,34 @@
+    private User user;
+
+    @DBRef
+    private OrderItem orderItem;
```
List<OrderItem> items?

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/model/OrderDetails.java:
```diff
@@ -0,0 +1,34 @@
+@Data
+@Builder
+@Document(collection = "OrderDetails")
+public class OrderDetails {
```
maybe will be better to merge order and orderDetails if we use mongodb? in nosql databases normalization is not required

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/model/AddressDelivery.java:
```diff
@@ -0,0 +1,19 @@
+    private String street;
+    private String house;
+    private Integer apartment;
+    private Integer comment;
```
Maybe it should be String not Integer?

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/controller/DeliveryController.java:
```diff
@@ -0,0 +1,59 @@
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = SwaggerResponse.Code.CODE_200, description = SwaggerResponse.Message.CODE_200_FOUND_DESCRIPTION) })
+    @PostMapping
+    public ResponseEntity<DeliveryCompanyAdminDto> saveNewDeliveryCompanies(@RequestParam @ValidName String companyName) {
```
saveNewDeliveryCompany

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/controller/DeliveryController.java:
```diff
@@ -0,0 +1,59 @@
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = SwaggerResponse.Code.CODE_200, description = SwaggerResponse.Message.CODE_200_FOUND_DESCRIPTION) })
+    @DeleteMapping
+    public ResponseEntity<DeliveryCompanyAdminDto> deleteDeliveryCompanies(@RequestParam @ValidName String companyName) {
```
deleteDeliveryCompany

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/controller/OrderController.java:
```diff
@@ -0,0 +1,45 @@
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = SwaggerResponse.Code.CODE_200, description = SwaggerResponse.Message.CODE_200_FOUND_DESCRIPTION) })
+    @PostMapping
+    public ResponseEntity<OrderNumberDto> newOrder(@RequestBody OrderDto orderDto, HttpServletRequest request) {
```
"createOrder" or "saveOrder", the method should be a verb

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/impl/DeliveryServiceImpl.java:
```diff
@@ -0,0 +1,59 @@
+        List<DeliveryCompany> deliveryCompanies = deliveryCompanyRepository.findAllByStatus(status);
+        if (deliveryCompanies.isEmpty()) {
+            return new ArrayList<>();
+        }
```
what need to return empty collection if collection is empty? :)

NatalliaBulyha replied:
мы не можем отправить deliveryCompanies, потому что контроллер ждет DeliveryCompanyDto, нужно мапить, на мой взгляд return new ArrayList<>() более емко говорит что мы отправляем пустой список

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,75 @@
+                .orderItem(orderItems)
+                .orderStatus(OrderStatus.NEW)
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
```
should we fill this field when create a new order?

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/model/OrderNumber.java:
```diff
@@ -0,0 +1,21 @@
+@Builder
+@Document(collection = "OrderNumber")
+public class OrderNumber {
+    private ObjectId id;
```
can we replace OrderNumber to order?

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,75 @@
+                .modifiedAt(LocalDateTime.now())
+                .build();
+
+        if (orderDto.getDelivery().getStreet() == null) {
```
more obvious to check if (deliveryCompanyName != null)

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/DeliveryBuilder.java:
```diff
@@ -0,0 +1,57 @@
+                .city(deliveryDto.getCity())
+                .paymentMethod(deliveryDto.getPaymentMethod())
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
```
should we fill this field during creating a delivery?

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/DeliveryBuilder.java:
```diff
@@ -0,0 +1,57 @@
+                .city(deliveryDto.getCity())
+                .paymentMethod(deliveryDto.getPaymentMethod())
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
```
should we fill this field during creating a delivery?

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/OrderItemBuilder.java:
```diff
@@ -0,0 +1,48 @@
+                        "doesn't found", orderItemDto.getProductSkuCode(), orderItemDto.getColorHex())));
+
+        if (inventory.getQuantity() < orderItemDto.getQuantity()) {
+         throw new ProductOutOfStockException("Product with sku code %s and color hex %s is out of stock.");
```
it also can be "Not enough quantity to order", out of stock is when a quantity is 0.

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/OrderItemBuilder.java:
```diff
@@ -0,0 +1,48 @@
+        //inventory.setQuantity(2000);
+        inventoryRepository.save(inventory);
+        OrderItem orderItem = OrderItem.builder()
+                .productColorId(inventory.getId())
```
maybe "inventory.getProductColor().getId()"?

NatalliaBulyha replied:
done

Commit:
CH-675/CH-676 validation

jira[endpoint: create order] https://cozyhometeam.atlassian.net/browse/CH-677?atlOrigin=eyJpIjoiNDYwMjQzYmI1MDkwNGRjOTllZGQwZTRmNjNkNGJiMWUiLCJwIjoiaiJ9

Commit:
CH-675/CH-676 Editing code by reviews

jira[endpoint: create order] https://cozyhometeam.atlassian.net/browse/CH-677?atlOrigin=eyJpIjoiNDYwMjQzYmI1MDkwNGRjOTllZGQwZTRmNjNkNGJiMWUiLCJwIjoiaiJ9

Commit:
CH-675/CH-676 added validation

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/JWT/JwtTokenFilter.java:
```diff
@@ -23,45 +23,43 @@
     private String userRoleAttributeName;
-    @Value("${api.basePath}")
+    @Value("${api.secure.basePath}")
     private String noAuthPathUrl;
```
authPathUrl

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/JWT/JwtTokenFilter.java:
```diff
@@ -23,45 +23,43 @@
-                log.info("[ON doFilterInternal]:: username [ {} ]", username);
+        String jwtToken = jwtTokenUtil.resolveToken(request);
+        String username = null;
+        if (shouldFilter(request) || jwtToken != null) {
```
good idea!)

Commit:
CH-675/CH-676 added logic to modifiedAt field

NatalliaBulyha committed:
Merge branch 'main' into CH-675/CH-676

Commit:
CH-675/CH-676 added check that the status is not DELETED

Commit:
Merge remote-tracking branch 'origin/CH-675/CH-676' into CH-675/CH-676

# Conflicts:
#	src/main/java/com/cozyhome/onlineshop/userservice/repository/UserRepository.java
#	src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserDetailsServiceImpl.java

Commit:
CH-675/CH-676 resolve conflicts

NatalliaBulyha committed:
Merge branch 'main' into CH-675/CH-676
